references.go

```diff
@@ -104,6 +104,16 @@ func walkGraph(result *[]*object.Commit, seen *map[plumbing.Hash]struct{}, curre
 	default: // more than one parent contains the path
 		// TODO: detect merges that had a conflict, because they must be
 		// included in the result here.
+		
```

Comment:
Most of `case 1` and `default` are the same. This switch could be simplified with:

```go
	case 0:
		*result = append(*result, current)
		return nil
	default: // or or more than one parent contains the path
		different, err := differentContents(path, current, parents)
		if err != nil {
			return err
		}
		if len(different) > 0 {
			*result = append(*result, current)
		}

		for _, p := range parents {
			err := walkGraph(result, seen, p, path)
			if err != nil {
				return err
			}
		}
	}
```